common/src/gadget/substrate/mod.rs

```diff
@@ -0,0 +1,299 @@
+use std::ops::Deref;
```

Comment:
I think everything in `gadget/substrate` should be moved out of common. It should be pulled into a directory called `gadget-impls` or something with subpackages that each implement connections to respective trait impls for this new API. Or at the very least, modularize everything into new packages. This is the "modular" in our modular restaking infrastructure that we need to keep driving home. CC @shekohex need to also ensure this matches our checklist module/breakdown for gadget-sdk.